pkg/3scale/amp/operator/highavailability_options_provider.go

```diff
@@ -27,24 +27,29 @@ func NewHighAvailabilityOptionsProvider(apimanager *appsv1alpha1.APIManager, nam
 }
 
 func (h *HighAvailabilityOptionsProvider) GetHighAvailabilityOptions() (*component.HighAvailabilityOptions, error) {
-	err := h.setBackendRedisOptions()
-	if err != nil {
-		return nil, err
+	setOptionsFns := []func(*HighAvailabilityOptionsProvider) error{}
```

Comment:
Just for curiosity, it the following approach equivalent?

```go
setOptionsFns := []func() error{}                                         
                                                                          
if h.apimanager.IsExternal(appsv1alpha1.BackendRedis) {                   
    setOptionsFns = append(setOptionsFns, h.setBackendRedisOptions)       
}                                                                         
if h.apimanager.IsExternal(appsv1alpha1.SystemRedis) {                    
    setOptionsFns = append(setOptionsFns, h.setSystemRedisOptions)        
}                                                                         
if h.apimanager.IsExternal(appsv1alpha1.SystemDatabase) {                 
    setOptionsFns = append(setOptionsFns, h.setSystemDatabaseOptions)     
}                                                                         
                                                                          
for _, setOptions := range setOptionsFns {                                
    if err := setOptions(); err != nil {                                  
        return nil, err                                                   
    }                                                                     
}                                                                         
```
In case they are equivalents,  does you approach have any added benefit?